bindings.c

```diff
@@ -2905,11 +2910,11 @@ int cg_rmdir(const char *path)
 
 	controller = pick_controller_from_path(fc, path);
 	if (!controller)
-		return -errno;
+		return errno == ENOENT ? -EPERM : -errno;
```

Comment:
I actually tend to disagree with my former consent to your implicit `-EPERM` objection. :)
We should be able to safely return -EPERM for:

```
 	if (!pick_controller_from_path())
                /* Someone's trying to delete "/cgroup". */

 	if (!find_cgroup_in_path())
 	        /* Someone's trying to delete a controller e.g. "/blkio". */

 	if (!get_cgdir_and_path()) {
 		/* Someone's trying to delete a cgroup on the same level as the
 		 * "/lxc" cgroup e.g. rmdir "/cgroup/blkio/lxc" or
 		 * rmdir "/cgroup/blkio/init.slice".
 		 */
 	}
```
All other interesting cases are caught further down.